sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultRoleDefinition.java

```diff
@@ -12,22 +12,23 @@
 public final class KeyVaultRoleDefinition {
     private final String id;
     private final String name;
-    private final String type;
+    private final KeyVaultRoleDefinitionType roleDefinitionType;
     private final KeyVaultRoleDefinitionProperties properties;
 
     /**
      * Creates a new {@link KeyVaultRoleDefinition role definition} with the specified details.
      *
      * @param id The ID for this {@link KeyVaultRoleDefinition role definition}.
      * @param name The name for this {@link KeyVaultRoleDefinition role definition}.
-     * @param type The type for this {@link KeyVaultRoleDefinition role definition}.
+     * @param roleDefinitionType The roleDefinitionType for this {@link KeyVaultRoleDefinition role definition}.
      * @param properties {@link KeyVaultRoleDefinitionProperties properties} of this {@link KeyVaultRoleDefinition
      * role definition}.
      */
-    public KeyVaultRoleDefinition(String id, String name, String type, KeyVaultRoleDefinitionProperties properties) {
```

Comment:
Just `type` is what we had decided on. The `role*` names in properties will differentiate even if you do flatten like we did in .NET, though I'm not sure why. @christothes?